### PR TITLE
Adds support for folder providers config in new file manager.

### PR DESCRIPTION
**Pull request by valadas**
- Adds a React control to list and remove folder providers configs
- Hooks into existing webforms settings control to implement 3rd-party configuration UI for each folder provider.

### CODE REVIEW

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:412]: DI guys @bdukes @ahoefling if you have a better idea of how to access EditUrl (which lives in PortalModuleBase) from a WebApi method, I am all ears on improving this :)

File: DNN Platform/Modules/ResourceManager/App_LocalResources/EditFolderMapping.ascx.resx
Comment (david-poindexter) [inline DNN Platform/Modules/ResourceManager/App_LocalResources/EditFolderMapping.ascx.resx:154]: @valadas does this follow the pattern used elsewhere?

File: DNN Platform/Modules/ResourceManager/App_LocalResources/ResourceManager.resx
Comment (david-poindexter) [inline DNN Platform/Modules/ResourceManager/App_LocalResources/ResourceManager.resx:295]:
```suggestion
    <value>Edit Folder Type</value>
```

File: DNN Platform/Modules/ResourceManager/App_LocalResources/ResourceManager.resx
Comment (david-poindexter) [inline DNN Platform/Modules/ResourceManager/App_LocalResources/ResourceManager.resx:304]:
```suggestion
    <value>Remove Folder Type</value>
```

**Review by david-poindexter — CHANGES_REQUESTED** [review]
Great work on this @valadas - we are getting closer and closer - woo hoo!  🎉

File: DNN Platform/Modules/ResourceManager/App_LocalResources/EditFolderMapping.ascx.resx
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/App_LocalResources/EditFolderMapping.ascx.resx:154]: It was copied directly from the old one... You have any suggestion here?

Comment (david-poindexter) [reply DNN Platform/Modules/ResourceManager/App_LocalResources/EditFolderMapping.ascx.resx:154]: No, I just didn't compare - I was being lazy I suppose.  😆

File: DNN Platform/Modules/ResourceManager/App_LocalResources/ResourceManager.resx
Comment (david-poindexter) [inline DNN Platform/Modules/ResourceManager/App_LocalResources/ResourceManager.resx:298]: Oops - I missed one.

```suggestion
    <value>Remove Folder Type</value>
```

**Review by david-poindexter — CHANGES_REQUESTED** [review]

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:412]: I think there is something missing if we want to access the `EditUrl`. We may need to add an `IModuleNavigationManager` to manage routes at the module level. This control would be similar to the `INavigaitonManager` but would take a parameter of ModuleId.

Another option, I don't recommend but it would be a solution to get through referencing the deprecated `ModuleControlFactory`. You could inject the `ModuleControlPipeline` which is the new version of that class and access the same `CreateModuleControl()` method.

It could look something like this
```C#
public class ItemController : DnnApiController
{
    // omitted code

    private readonly IModuleControlPipeline modulePipeline;

    public ItemController(IModuleControlPipeline modulePipeline)
    {
        this.modulePipeline = modulePipeline;
    } 

    // omitted code

    private ModuleInstanceContext GetModuleContext()
    {
        IModuleControl moduleControl = modulePipeline.CreateModuleControl(this.ActiveModule);
        return new ModuleInstanceContext(moduleControl);
    }

    // omitted code
}
```

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (ahoefling) [inline DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:62]: Could this be added to the `NavigationManager`?

**Review by ahoefling — CHANGES_REQUESTED** [review]
I left a review as @valadas requested. I know I am not a core reviewer, but I thought I would leave some notes across the entire Pull Request. Overall this looks great and thanks for the contribution!

I left a bunch of comments on coding style that will generate build warnings and we should be striving to reduce those as much as possible.

Dependency Injection in DNN is starting to be used in many 3rd party modules.  But, there is not enough support for Dependency Injection across the core library making changes like this difficult. My personal approach is identifying tangible pieces to refactor into interfaces and submitting Pull Requests. I believe we have found some pieces, most importantly managing Module Navigation. I don't think it should hold up a PR like this, but this feature won't be ready for v11 as the `ModuleControlFactory` is planned for deprecation. 

Here are a couple areas in DNN that I identified in this PR that could be good candidates for new abstractions for DI
- ThumbnailManager
- ModuleNavigationManager
- FolderMappingController

**Comment by valadas** [comment]
Thank you @ahoefling 
I will look into all that in more details a bit later today.

I think we are missing the stylecop config on this project, is it just a matter of copying the stylecop.json file into the project ? I would be happy to rewrite those things to respect the new code rules.

The webforms control was a direct copy from the other file manager, I only adjusted a couple of lines for proper urls and removing the telerik dropdown. But this also I would be happy to correct with the new stylecop config in place... I'll correct this PR and then probably submit another one with stylecop in place...

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: Excellent, works like a charm and implemented in this PR.

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs]: I have renamed the 3 private variables in that file.

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs]: Done

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:111]: There are many of these small issue, I think I will leave this PR as is, let's merge it and then I will add the stylecop config and do a separate PR to fix here and other files of this branch... It will be a smaller review that way I think...

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:116]: Will do a separate PR for this as the previous comment

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs]: Done

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:39]: Done

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: Done

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:64]: I am not sure what you mean by this ? @ahoefling

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: Glad this worked

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:64]: The Dnn `NavigationManager` manages several common routing scenarios in DNN. It doesn't include all of them, but is at a great starting point. The `Globals.AccessDeniedURL()` really should be ported into the `NavigationManager`. 

I propose the following changes
- Add new API to `INavigationManager` - `AccessDeniedUrl()`
- Implement new API in `NavigationManager` by porting the `Globals.AccessDeniedURL()` code
- Update the `Globals.AccessDeniedURL()` to use the `INavigationManager` new API

This may be a bit to refactor, it is sometimes hard to tell just by looking at the API structure. @bdukes or @mitchelsellers do you have any thoughts on this.

Comment (valadas) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:64]: Ha, I understand now :)

**Comment by valadas** [comment]
As discussed on phone, we are merging this to unblock other in progress work on this branch, we will have to re-review the actual whole project anyway later when we merge this into develop.

File: DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs
Comment (bdukes) [reply DNN Platform/Modules/ResourceManager/EditFolderMapping.ascx.cs:64]: Yeah, moving more navigation methods to `INavigationManager` would be nice.  However, we need to be thoughtful that adding members to an interface is a breaking change, since someone may be trying to provide an alternate implementation (at the very least via tests).

**Comment by bdukes** [comment]
@valadas In Visual Studio, copy the `stylecop.json` file from an existing project into this new project.  It should show up in the project file as a link, rather than an entirely new file (there's only one `stylecop.json` file, and all of the projects reference it)

**Pull request merged**
